pydrive2/files.py

```diff
@@ -518,6 +518,40 @@ def Delete(self, param=None):
         """
         self._FilesDelete(param=param)
 
+    @LoadAuth
+    def Copy(self, target_folder=None, new_title=None, param=None):
+        """Creates a copy of this file. Folders cannot be copied.
+
+        :param target_folder: Folder where the file will be copied.
+        :type target_folder: GoogleDriveFile
+        :param new_title: Name of the new file., defaults to None
+        :type new_title: str, optional
+        :param param: addition parameters to pass, defaults to None
+        :type param: dict, optional
+        :raises ApiRequestError
+        :return: the copied file
+        :rtype: GoogleDriveFile
+        """        
+
+        if param is None:
+            param = {}
+
+        param["fileId"] = self["id"]
+        param["supportsAllDrives"] = True
+        param["body"] = {}
+
+        if target_folder:
+            param["body"]["parents"] = [{"id": target_folder["id"]}]
```

Comment:
q: can you pass multiple parents?

Reply:
apparently no:

```python
[WARNING] googleapiclient.http -> _should_retry_response: Encountered 403 Forbidden with reason "cannotAddParent"
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 552, in Copy
    self.auth.service.files().copy(**param).execute(http=self.http)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/http.py", line 938, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1WO_CTrDj62v3RWJ5OP5vCpsNlCKF38mIPQ-nAcog8H0/copy?supportsAllDrives=true&alt=json returned "Increasing the number of parents is not allowed". Details: "[{'domain': 'global', 'reason': 'cannotAddParent', 'message': 'Increasing the number of parents is not allowed'}]">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/main.py", line 71, in <module>
    f = file.Copy(folders)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/auth.py", line 82, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 555, in Copy
    raise ApiRequestError(error)
pydrive2.files.ApiRequestError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1WO_CTrDj62v3RWJ5OP5vCpsNlCKF38mIPQ-nAcog8H0/copy?supportsAllDrives=true&alt=json returned "Increasing the number of parents is not allowed". Details: "[{'domain': 'global', 'reason': 'cannotAddParent', 'message': 'Increasing the number of parents is not allowed'}]">
```

maybe this is allowed in some other types of Drive, otherwise it doesn't make sent that `parents` is an array if it can have only one element.

Reply:
thanks!